src/gpu/nvidia/sycl_cuda_compat.hpp

```diff
@@ -35,9 +35,13 @@ T get_native_mem(const interop_handle &ih, U acc) {
             ih.get_native_mem<::sycl::backend::ext_oneapi_cuda>(acc));
 }
 
-template <typename T>
-void host_task(::sycl::handler &cgh, const T &task) {
+template <typename HandlerT, typename FnT>
```

Comment:
The spec says:
> The call to interopCallable must not submit any synchronous tasks to the native backend object, and it must not block waiting for any tasks to complete. The call also must not add tasks to backend objects that underly any other queue, aside from the queue that is associated with this handler. If it does any of these things, the behavior is undefined.

And we have a few places where we do explicitly synchronize, [here](https://github.com/oneapi-src/oneDNN/blob/51b489fe514cb827de37ba28246ea435bc81cce7/src/gpu/nvidia/cudnn_batch_normalization_executor.hpp#L201), for example.

It seems that it may become an issue when `ext_codeplay_enqueue_native_command` is used.

Reply:
In validation the following case hung and the process was killed by timeout: `--mode-modifier=P --bnorm --engine=gpu --inplace=true ic23n"bnorm_ci_0d:1"`, so there seems to be an issue indeed. Given that we enable the new host task extension conditionally we should also introduce a a function that should do the synchronization under the same condition.

Reply:
I have made some changes and added a utility `sync_device` which would synchronize only when using default host_task.
I wanted to additionally ask, what is the reason for the additional interop_task? If my understanding is correct the current situation is ```q.submit{ host_task { q.submit { memset/device sync } } }``` in the current state I do not think that the internal submit will wait on any prior events, as that would be handled by the external submit, and no following dependencies will come out from the internal submit as we synchronize the device in the `host_task` case, while the `enqueue_native_command` would resolve any dependencies.

I have removed the internal interop task and as I am not seeing failures without it, let me know if this change does not make sense/ is wrong, and I can revert it.

Reply:
I don't know why there was the additional interop_task and I can't think of any reason why we may need it if we block the entire device anyway.